src/main/java/org/spongepowered/asm/mixin/transformer/MixinApplicatorInterface.java

```diff
@@ -101,23 +107,29 @@ protected void prepareInjections(MixinTargetContext mixin) {
             try {
                 InjectionInfo injectInfo = InjectionInfo.parse(mixin, method);
                 if (injectInfo != null) {
-                    throw new InvalidInterfaceMixinException(mixin, injectInfo + " is not supported on interface mixin method " + method.name);
+                    if (!Bytecode.hasFlag(method, Opcodes.ACC_STATIC) && !MixinEnvironment.getCompatibilityLevel().isAtLeast(MixinEnvironment.CompatibilityLevel.JAVA_8)) {
```

Comment:
Does this actually support default methods ?

Reply:
Yes, as long as Mixin is configured to run on, or running on, Java 8+. We don't want to accidentally allow inserting default methods where they're unsupported.

Reply:
Oh, you actually allow inserting into any interface method when on Java 8. Seems good to me.